keras/layers/preprocessing/audio_preprocessing.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Keras audio preprocessing layers."""
+
+import math
+
+import tensorflow.compat.v2 as tf
+from tensorflow.python.util.tf_export import keras_export
+
+from keras.engine import base_layer
+from keras.engine import base_preprocessing_layer
+from keras.layers.preprocessing import preprocessing_utils as utils
+
+
+@keras_export(
+    "keras.layers.MelSpectrogram",
+    "keras.layers.experimental.preprocessing.MelSpectrogram",
+)
+class MelSpectrogram(base_layer.Layer):
+    """A preprocessing layer to convert raw audio signals to Mel spectrograms.
+
+    This layer takes `float32/float64` single or batched audio signal as inputs
+    and computes the Mel spectrogram using Short-Time Fourier Transform (STFT)
+    and Mel scaling. The input should be a 1D (unbatched) or 2D (batched) tensor
+    representing audio signals. The output will be a 2D or 3D tensor
+    representing Mel spectrograms.
+
+    For an overview and full list of preprocessing layers, see the preprocessing
+    [guide](https://www.tensorflow.org/guide/keras/preprocessing_layers).
+
+    Input shape:
+        1D (unbatched) or 2D (batched) tensor with shape:`(..., samples)`.
+
+    Output shape:
+        2D (unbatched) or 3D (batched) tensor with shape:`(..., n_mels, time)`.
+
+    Args:
+        n_fft: Integer, size of the FFT window.
+        hop_length: Integer, number of samples between successive STFT columns.
+        win_length: Integer, size of the STFT window.
+            If `None`, defaults to `n_fft`.
+        window_fn: String, name of the window function to use.
+        sr: Integer, sample rate of the input signal.
+        n_mels: Integer, number of mel bins to generate.
+        fmin: Float, minimum frequency of the mel bins.
+        fmax: Float, maximum frequency of the mel bins.
+            If `None`, defaults to `sr / 2`.
+        power_to_db: If True, convert the power spectrogram to decibels.
+        top_db: Float, minimum negative cut-off `max(10 * log10(S)) - top_db`.
+        power: Float, exponent for the magnitude spectrogram.
+            1 for magnitude, 2 for power, etc. Default is 2.
+        ref: Float, the power is scaled relative to it `10 * log10(S / ref)`.
+        amin: Float, minimum value for power and `ref`.
+    """
```

Comment:
Please add a simple code example in the docstring, for the most common use case.